subsys/bluetooth/controller/ll_sw/ull_adv_sync.c

```diff
@@ -378,11 +378,27 @@ uint8_t ll_adv_sync_param_set(uint8_t handle, uint16_t interval, uint16_t flags)
 
 	sync->interval = interval;
 
-	err = ull_adv_sync_pdu_set_clear(adv, 0, 0, NULL, &ter_idx);
+	err = ull_adv_sync_pdu_set_clear(adv, ULL_ADV_PDU_HDR_FIELD_AUX_PTR, 0, NULL, &ter_idx);
 	if (err) {
 		return err;
 	}
 
+	do {
+		const char *tmp = "test test test";
+		struct pdu_adv *test_pdu;
+		struct pdu_adv *ter_pdu;
+
+		/* Allocate new PDU */
+		test_pdu = lll_adv_pdu_alloc_pdu_adv();
+		/* Initialize new empty PDU */
+		adv_sync_pdu_init(test_pdu, 0);
+		/* Add some AD for testing */
+		adv_sync_pdu_ad_data_set(test_pdu, tmp, strlen(tmp));
+		/* Link to 1st PDU */
+		ter_pdu = (void *)lll_sync->data.pdu[ter_idx];
+		lll_adv_pdu_linked_append(test_pdu, ter_pdu);
+	} while (0);
```

Comment:
@ppryga This is similar to what (assuming my proposal makes sense) needs to be done when adding CTEInfo: need to allocate new empty PDU, initialize it with CTEInfo header (2nd parameter to adv_sync_pdu_init) and optionally AuxPtr (for all but last PDU), add CTEInfo (adv_sync_cte_info_set) and then link it to parent PDU. Such chain will be preserved if AD is changed, i.e. it will be duplicated as-is.

Reply:
@andrzej-kaczmarek I'd like try to write small test code that will check this approach. I'll try to do it in the beginning of next week.